src/solver/brent/mod.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018-2020 argmin developers
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! Brent's method
+//!
+//! A root-finding algorithm combining the bisection method, the secant method
+//! and inverse quadratic interpolation. It has the reliability of bisection
+//! but it can be as quick as some of the less-reliable methods.
+//!
+//! # References:
+//!
+//! https://en.wikipedia.org/wiki/Brent%27s_method
+//!
+
+/// Implementation of Brent's optimization method,
+/// see https://en.wikipedia.org/wiki/Brent%27s_method
+use crate::prelude::*;
+use serde::{Deserialize, Serialize};
+use std::f64;
+use std::fmt;
+
+/// Error to be thrown if Brent is initialized with improper parameters.
+#[derive(Debug)]
+pub struct BrentError;
+
+impl fmt::Display for BrentError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Brent error: f(min) and f(max) must have different signs."
+        )
+    }
+}
+
+impl std::error::Error for BrentError {
+    fn description(&self) -> &str {
+        "Brent requires a min and max value bracketing the root of the function."
+    }
+}
+
+/// Brent's method
+///
+/// A root-finding algorithm combining the bisection method, the secant method
+/// and inverse quadratic interpolation. It has the reliability of bisection
+/// but it can be as quick as some of the less-reliable methods.
+///
+/// # References:
+/// https://en.wikipedia.org/wiki/Brent%27s_method
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Brent {
```

Comment:
Could you please comment the individual fields?

Reply:
I have added the comments.